docs/DeveloperGuide.md

```diff
@@ -1400,31 +1304,36 @@ testers are expected to do more *exploratory* testing.
    1. Other incorrect delete tag commands to try: `delete_tag`, `delete_tag x`, `...`<br>
       Expected: Similar to previous.
 
-1. Deleting a tag while on the home page
+### Filtering contacts by tag
+1. Filtering contacts while all contacts are being shown
 
-   1. Prerequisites: Be on the home screen using the `home` command.
+   1. Prerequisites: List all contacts using the `view_contacts` command. Multiple contacts in the list.
 
-   1. Test case: `delete_tag t/clients`<br>
-      Expected: `clients` tag is deleted. Contacts having the `clients` tag has the tag removed from the list. Details of the deleted tag shown in the status message.
+   2. Test case: `filter_by_tag vendors`<br>
+      Expected: All contacts with the tag `vendors` are shown. Command success status message shown with the number of contacts listed (can be 0 if no valid contacts found).
 
-   1. Test case: `delete_tag t/`<br>
-      Expected: No tag is deleted. Error details shown in the status message.
+   3. Test case: `filter_by_tag vendors customers`<br>
+      Expected: All contacts with the tag `vendors` and all contacts with the tag `customers` are shown. Command success status message shown with the number of contacts listed (can be 0 if no valid contacts found).
 
-   1. Other incorrect delete tag commands to try: `delete_tag`, `delete_tag x`, `...`<br>
+   4. Test case: `filter_by_tag`<br>
+      Expected: Error details shown in the status message.
+
+   5. Other incorrect filter contact commands to try: `filter_by_tag @`, `...`
       Expected: Similar to previous.
 
-1. Deleting a tag after selecting an event
+2. Filtering contacts while on the home page
 
-   1. Prerequisites: Be on the selected event page using the `select_event x` command in which x represents an integer that is smaller than or equal to the event list size.
+   1. Prerequisites: Be on the home screen using the `home` command. Contact list should be displayed in the middle.
 
-   1. Test case: `delete_tag t/clients`<br>
-      Expected: Deletes the `clients` tag. Returns back to home page, and reflects the changes in the entire contact list. Details of the deleted tag shown in the status message.
+   1. Test cases and expected outcomes are similar to the previous case.
 
-   1. Test case: `delete_tag t/`<br>
-      Expected: No tag is deleted. Error details shown in the status message.
+3. Filtering contacts after selecting an event
 
-   1. Other incorrect delete tag commands to try: `delete_tag`, `delete_tag x`, `...`<br>
-      Expected: Similar to previous.
+   1. Prerequisites: Be on the selected event page using the `select_event x` command in which x represents a positive integer that is smaller than or equal to the event list size. Multiple contacts in the list of contacts linked to the event.
```

Comment:
perhaps could state x in `select_event x` refers to index of the list displayed instead?

Reply:
```suggestion
   1. Prerequisites: Be on the selected event page using the `select_event x` command in which x represents a positive integer that is smaller than or equal to the event list size. There must be contacts linked to the event.
```